mobile/native/src/ln_dlc/node.rs

```diff
@@ -36,78 +33,107 @@ impl Node {
                     );
-                    dlc_message_handler.send_message(node_id, Message::SubChannel(msg.clone()));
+                    let reply_msg = self
+                        .inner
+                        .sub_channel_manager
+                        .on_sub_channel_message(&incoming_msg, &node_id)
+                        .map_err(|e| anyhow!(e.to_string()))?;
 
-                    if let SubChannelMessage::Finalize(_) = msg {
-                        let offer_collateral =
-                            get_first_confirmed_dlc(dlc_manager)?.offer_collateral;
+                    if let SubChannelMessage::Offer(offer) = &incoming_msg {
+                        let channel_id = offer.channel_id;
 
-                        let filled_order = match order::handler::order_filled() {
-                            Ok(filled_order) => filled_order,
-                            Err(e) => {
-                                tracing::error!("Critical Error! We have a DLC but were unable to set the order to filled: {e:#}");
-                                continue;
-                            }
-                        };
+                        // TODO: We should probably verify that: (1) the counterparty is the
+                        // coordinator and (2) the DLC channel offer is expected and correct.
+                        if let Err(e) = self.inner.accept_dlc_channel_offer(&channel_id) {
+                            tracing::error!(channel_id = %hex::encode(channel_id), "Failed to accept DLC channel offer: {e:#}");
+                        }
+                    }
 
-                        if let Err(e) =
-                            position::handler::order_filled(filled_order, offer_collateral)
+                    if let SubChannelMessage::CloseOffer(offer) = &incoming_msg {
+                        let channel_id = offer.channel_id;
+
+                        // TODO: We should probably verify that: (1) the counterparty is the
+                        // coordinator and (2) the DLC channel close offer is expected and correct.
+                        if let Err(e) = self
+                            .inner
+                            .accept_dlc_channel_collaborative_settlement(&channel_id)
                         {
-                            tracing::error!("Failed to handle position after receiving DLC: {e:#}");
-                            continue;
+                            tracing::error!(channel_id = %hex::encode(channel_id), "Failed to accept DLC channel close offer: {e:#}");
+                        }
+                    }
+
+                    if let Some(reply_msg) = reply_msg {
+                        tracing::debug!(
+                            to = %node_id,
+                            msg = %sub_channel_message_as_str(&reply_msg),
+                            "Sending DLC channel message"
+                        );
+                        self.inner
```

Comment:
side note, we will have to delay the sending of the confirm messages here.

Reply:
Confirm is sent by the coordinator though.